numpy 1.19.0 + intelmkl test error

python-3.8.3, numpy 1.19.0 

numpy is compiled against intel mkl 2018.2.199 (newer version of mkl + numpy => nvidia's nvblas no longer work )

OS ubuntu linux.
```
np.show_config()
blas_mkl_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/opt/intel/mkl/lib/intel64/']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/opt/intel/mkl/include']
blas_opt_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/opt/intel/mkl/lib/intel64/']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/opt/intel/mkl/include']
lapack_mkl_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/opt/intel/mkl/lib/intel64/']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/opt/intel/mkl/include']
lapack_opt_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/opt/intel/mkl/lib/intel64/']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/opt/intel/mkl/include']
>>> np.__version__
'1.19.0'
```

```
TestSystemInfoReading.test_overrides _____________________

self = <numpy.distutils.tests.test_system_info.TestSystemInfoReading object at 0x7f36d855d640>

    def test_overrides(self):
        previousDir = os.getcwd()
        cfg = os.path.join(self._dir1, 'site.cfg')
        shutil.copy(self._sitecfg, cfg)
        try:
            os.chdir(self._dir1)
            # Check that the '[ALL]' section does not override
            # missing values from other sections
            info = mkl_info()
            lib_dirs = info.cp['ALL']['library_dirs'].split(os.pathsep)
            assert info.get_lib_dirs() != lib_dirs
    
            # But if we copy the values to a '[mkl]' section the value
            # is correct
            with open(cfg, 'r') as fid:
                mkl = fid.read().replace('ALL', 'mkl')
            with open(cfg, 'w') as fid:
                fid.write(mkl)
            info = mkl_info()
            assert info.get_lib_dirs() == lib_dirs
    
            # Also, the values will be taken from a section named '[DEFAULT]'
            with open(cfg, 'r') as fid:
                dflt = fid.read().replace('mkl', 'DEFAULT')
            with open(cfg, 'w') as fid:
                fid.write(dflt)
            info = mkl_info()
>           assert info.get_lib_dirs() == lib_dirs
E           AssertionError: assert ['/opt/intel/...lib/intel64/'] == ['/tmp/tmp_gt.../tmp0ntuctr2']
E             At index 0 diff: '/opt/intel/mkl/lib/intel64/' != '/tmp/tmp_gtgtt2a'
E             Right contains one more item: '/tmp/tmp0ntuctr2'
E             Full diff:
E             - ['/tmp/tmp_gtgtt2a', '/tmp/tmp0ntuctr2']
E             + ['/opt/intel/mkl/lib/intel64/']

cfg        = '/tmp/tmp_gtgtt2a/site.cfg'
dflt       = '\n[DEFAULT]\nlibrary_dirs = /tmp/tmp_gtgtt2a:/tmp/tmp0ntuctr2\nlibraries = /tmp/tmp_gtgtt2a/libfoo.so,/tmp/tmp0ntuctr.../tmp0ntuctr2\n\n[duplicate_options]\nmylib_libs = /tmp/tmp_gtgtt2a/libfoo.so\nlibraries = /tmp/tmp0ntuctr2/libbar.so\n'
fid        = <_io.TextIOWrapper name='/tmp/tmp_gtgtt2a/site.cfg' mode='w' encoding='UTF-8'>
info       = <numpy.distutils.system_info.mkl_info object at 0x7f36d8570be0>
lib_dirs   = ['/tmp/tmp_gtgtt2a', '/tmp/tmp0ntuctr2']
mkl        = '\n[mkl]\nlibrary_dirs = /tmp/tmp_gtgtt2a:/tmp/tmp0ntuctr2\nlibraries = /tmp/tmp_gtgtt2a/libfoo.so,/tmp/tmp0ntuctr2/li.../tmp0ntuctr2\n\n[duplicate_options]\nmylib_libs = /tmp/tmp_gtgtt2a/libfoo.so\nlibraries = /tmp/tmp0ntuctr2/libbar.so\n'
previousDir = '/home/bernard/opt/python38'
self       = <numpy.distutils.tests.test_system_info.TestSystemInfoReading object at 0x7f36d855d640>

lib/python3.8/site-packages/numpy/distutils/tests/test_system_info.py:284: AssertionError
```

Similar error:
```
_________________________________________________ TestSystemInfoReading.test_overrides _________________________________________________

self = <numpy.distutils.tests.test_system_info.TestSystemInfoReading object at 0x7fd9e56d5be0>

    def test_overrides(self):
        previousDir = os.getcwd()
        cfg = os.path.join(self._dir1, 'site.cfg')
        shutil.copy(self._sitecfg, cfg)
        try:
            os.chdir(self._dir1)
            # Check that the '[ALL]' section does not override
            # missing values from other sections
            info = mkl_info()
            lib_dirs = info.cp['ALL']['library_dirs'].split(os.pathsep)
            assert info.get_lib_dirs() != lib_dirs
    
            # But if we copy the values to a '[mkl]' section the value
            # is correct
            with open(cfg, 'r') as fid:
                mkl = fid.read().replace('ALL', 'mkl')
            with open(cfg, 'w') as fid:
                fid.write(mkl)
            info = mkl_info()
>           assert info.get_lib_dirs() == lib_dirs
E           AssertionError: assert ['/opt/intel/.../tmpah6sjshj'] == ['/tmp/tmp9f5.../tmpah6sjshj']
E             At index 0 diff: '/opt/intel/compilers_and_libraries_2020.1.217/linux/mkl' != '/tmp/tmp9f5gbkuk'
E             Left contains 3 more items, first extra item: '/opt/intel/compilers_and_libraries_2020.1.217/linux/mkl/lib'
E             Full diff:
E               [
E             +  '/opt/intel/compilers_and_libraries_2020.1.217/linux/mkl',
E             +  '/opt/intel/compilers_and_libraries_2020.1.217/linux/mkl/include',
E             +  '/opt/intel/compilers_and_libraries_2020.1.217/linux/mkl/lib',...
E             
E             ...Full output truncated (4 lines hidden), use '-vv' to show

cfg        = '/tmp/tmp9f5gbkuk/site.cfg'
fid        = <_io.TextIOWrapper name='/tmp/tmp9f5gbkuk/site.cfg' mode='w' encoding='ANSI_X3.4-1968'>
info       = <numpy.distutils.system_info.mkl_info object at 0x7fd9e56e8f28>
lib_dirs   = ['/tmp/tmp9f5gbkuk', '/tmp/tmpah6sjshj']
mkl        = '\n[mkl]\nlibrary_dirs = /tmp/tmp9f5gbkuk:/tmp/tmpah6sjshj\nlibraries = /tmp/tmp9f5gbkuk/libfoo.so,/tmp/tmpah6sjshj/li.../tmpah6sjshj\n\n[duplicate_options]\nmylib_libs = /tmp/tmp9f5gbkuk/libfoo.so\nlibraries = /tmp/tmpah6sjshj/libbar.so\n'
previousDir = '/home/user/Work'
self       = <numpy.distutils.tests.test_system_info.TestSystemInfoReading object at 0x7fd9e56d5be0>

/usr/local/lib/python3.6/dist-packages/numpy/distutils/tests/test_system_info.py:276: AssertionError
```

I have done more experiments that may shed more light on the issue.

it seems that the error is due to having a ,numpy-site.cfg in the $HOME. I tried with numpy precompiled binary installed with pip and got the same error. But if rename  .numpy-site-cfg and run test again then this error disappeared. However the precompiled binary exhibit some strange behaviour

If linked to openblas at run time  via update-alternatives (with .numpy-site-cfg renamed) it produce this error:
```
 TestF77Mismatch.test_lapack ____________________________________________________________________________________________

self = <numpy.linalg.tests.test_build.TestF77Mismatch object at 0x7fcae7f62ed0>

        @pytest.mark.skipif(not(sys.platform[:5] == 'linux'),
                            reason="no fortran compiler on non-Linux platform")
        def test_lapack(self):
            f = FindDependenciesLdd()
            deps = f.grep_dependencies(lapack_lite.__file__,
                                       [b'libg2c', b'libgfortran'])
            assert_(len(deps) <= 1,
                             """Both g77 and gfortran runtimes linked in lapack_lite ! This is likely to
    cause random crashes and wrong results. See numpy INSTALL.txt for more
>   information.""")
E   AssertionError: Both g77 and gfortran runtimes linked in lapack_lite ! This is likely to
E   cause random crashes and wrong results. See numpy INSTALL.txt for more
E   information.

deps       = [b'libgfortran', b'libgfortran']
f          = <numpy.linalg.tests.test_build.FindDependenciesLdd object at 0x7fcae692ef10>
self       = <numpy.linalg.tests.test_build.TestF77Mismatch object at 0x7fcae7f62ed0>

../lib/python3.7/site-packages/numpy/linalg/tests/test_build.py:53: AssertionError
```

if linked to MKL at runtime (.numpy-site.cfg renamed) then all tests passed.

If compile numpy-1.19.1 from source like before, but rename  .nuumpy-site.cfg after compilation and install then run the tests then all tests passed whether linking to openblas or MKL at runtime

For self compiled numpy-1.19.1

numpy.show_config() still shows
```
np.show_config()
blas_mkl_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/opt/intel/mkl/lib/intel64/']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/opt/intel/mkl/include']
blas_opt_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/opt/intel/mkl/lib/intel64/']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/opt/intel/mkl/include']
lapack_mkl_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/opt/intel/mkl/lib/intel64/']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/opt/intel/mkl/include']
lapack_opt_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/opt/intel/mkl/lib/intel64/']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/opt/intel/mkl/include']
```
Here is my .numpy-site.cfg
```
#[ALL]
#ibrary_dirs = /usr/lib
#include_dirs = /usr/include


[mkl] 
library_dirs = /opt/intel/mkl/lib/intel64/
include_dirs = /opt/intel/mkl/include
mkl_libs = mkl_rt 
lapack_libs = 

```
Probably there is something wrong here.

--- a/numpy/distutils/tests/test_system_info.py
+++ b/numpy/distutils/tests/test_system_info.py
@@ -254,6 +254,10 @@ class TestSystemInfoReading:
         finally:
             os.chdir(previousDir)
 
+    HAS_MKL = "mkl_rt" in mkl_info().calc_libraries_info().get("libraries", [])
+
+    @pytest.mark.xfail(HAS_MKL, reason=("`[DEFAULT]` override doesn't work if "
+                                        "numpy is built with MKL support"))
     def test_overrides(self):
         previousDir = os.getcwd()
         cfg = os.path.join(self._dir1, 'site.cfg')

